src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishAnAotApp.cs

```diff
@@ -90,15 +89,52 @@ public void NativeAot_only_runs_when_switch_is_enabled(string targetFramework)
                     .Should().Pass()
                     .And.HaveStdOutContaining("warning IL3050");
 
-                var publishDirectory = publishCommand.GetOutputDirectory(targetFramework: targetFramework, runtimeIdentifier: rid).FullName;
+                var publishDirectory = publishCommand.GetOutputDirectory(targetFramework: targetFramework, runtimeIdentifier: rid);
 
-                var publishedExe = Path.Combine(publishDirectory, $"{projectName}.exe");
+                var publishedExe = Path.Combine(publishDirectory.FullName, $"{testProject.Name}{Constants.ExeSuffix}");
 
                 // The exe exist and should be native
                 File.Exists(publishedExe).Should().BeTrue();
                 IsNativeImage(publishedExe).Should().BeTrue();
+
+                var command = new RunExeCommand(Log, publishedExe)
+                    .Execute().Should().Pass()
+                    .And.HaveStdOutContaining("Hello world");                
             }
+        }
+
+        [RequiresMSBuildVersionTheory("17.0.0.32901")]
+        [InlineData(LatestTfm)]
+        public void NativeAotLib_only_runs_when_switch_is_enabled(string targetFramework)
+        {
+
+            // Need to enable this in a Linux distro by adding the pre-reqs to a suitable container
+            // https://github.com/dotnet/sdk/issues/24983
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```

Comment:
I think this comment is not relevant anymore since we already have the new container, meaning that this test could run for Linux too